extension/framework/runner/environment.go

```diff
@@ -277,14 +276,21 @@ func (env *Environment) mockFunction(functionName string) {
 				functionName, readableArguments, valueSliceToString(expectedArguments)))
 		}
 
-		response := responses[0]
-		responses = responses[1:]
-		env.setToOtto(functionName, "responses", responses)
+		var answer otto.Value
+		if len(exceptions) > 0 {
```

Comment:
this won't work. consider:
```
someCall().Expect().Return(1)
someCall().Expect().Return(2)
someCall().Expect().Throw(3)
```
The calls should be processed in order, currently the first call will throw.